### ISSUE TITLE
indent: Wrapped lines/line continuation indentation

### ISSUE THREAD

**Issue by saveman71** [eslint/eslint#12945]
**What rule do you want to change?**

`indent`

**Does this change cause the rule to produce more or fewer warnings?** 

More!

**How will the change be implemented? (New option, new default behavior, etc.)?**

I'm thinking a new option with a default behaviour to continue indentation at x + 1 unit of indentation.

**Please provide some example code that this change will affect:**

```js
        // Good example
        const longResultVariable =
          await MyInstance.getInfo(parameter, parameterOther, parameterLongLong);

        // But this is valid
        const otherExample = Array.isArray(longResultVariable) &&
        longResultVariable.length > 0;

        // And this too
        const evenThis = Array.isArray(isValid) &&
              isValid.length > 0;

        // And this too!
        const andEvenThat = Array.isArray(isValid) &&
  isValid.length > 0;
```

**What does the rule currently do for this code?**

Currently, the `indent` rule emits no warning.

**What will the rule do after it's changed?**

Only the first case would be valid.

**Are you willing to submit a pull request to implement this change?**

Maybe. I'm not very experimented with AST trees and indentation seems like a tricky subject.

**Comment by kaicataldo** [comment]
Duplicate of https://github.com/eslint/eslint/issues/12248.

**Comment by saveman71** [comment]
Thank you!
